src/common/helpers/data-extract/data-extract.js

```diff
@@ -99,38 +99,40 @@ export const getTbLicenceType = (application) => {
   const destinationType = getDestinationType(application)
 
   if (
-    originType?.value !== 'afu' &&
-    !isTbRestricted(originType) &&
-    isTbRestricted(destinationType)
+    (originType?.value === 'market' && isTbRestricted(destinationType)) ||
+    (originType?.value === 'unrestricted-farm' &&
+      isTbRestricted(destinationType)) ||
+    (originType?.value === 'after-import-location' &&
+      isTbRestricted(destinationType))
   ) {
     return 'TB15'
   }
 
-  if (isTbRestricted(originType) && isTbRestricted(destinationType)) {
-    return 'TB16'
-  }
-
-  if (isTbRestricted(originType) && destinationType?.value === 'afu') {
-    return 'TB16e'
-  }
-
   if (
     isTbRestricted(originType) &&
-    destinationType?.value === 'dedicated-sale'
+    isTbRestricted(destinationType) &&
+    destinationType?.value !== 'iso-unit'
   ) {
-    return 'TB16e'
+    return 'TB16'
   }
 
   if (
-    originType?.value === 'afu' &&
-    ['slaughter', 'afu', 'dedicated-sale'].includes(destinationType?.value)
+    (isTbRestricted(originType) &&
+      (destinationType?.value === 'dedicated-sale' ||
+        destinationType?.value === 'afu')) ||
+    (originType?.value === 'afu' &&
+      ['slaughter', 'afu', 'dedicated-sale'].includes(destinationType?.value))
   ) {
     return 'TB16e'
   }
 
-  if (isTbRestricted(originType) && destinationType?.value === 'slaughter') {
+  if (
+    isTbRestricted(originType) &&
+    ['slaughter', ''].includes(destinationType?.value)
```

Comment:
not sure why we need the '' case here, since this value comes from a mandatory set of radio buttons - we should always have a case